providers/common/ciphers/ccm.c

```diff
@@ -0,0 +1,529 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core_numbers.h>
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+#include <openssl/params.h>
+#include "internal/provider_algs.h"
+#include "internal/providercommonerr.h"
+#include "ciphers_locl.h"
+
+#define CCM_FLAGS (EVP_CIPH_FLAG_AEAD_CIPHER | EVP_CIPH_FLAG_DEFAULT_ASN1      \
+                   | EVP_CIPH_CUSTOM_IV | EVP_CIPH_FLAG_CUSTOM_CIPHER          \
+                   | EVP_CIPH_ALWAYS_CALL_INIT | EVP_CIPH_CTRL_INIT            \
+                   | EVP_CIPH_CUSTOM_COPY)
```

Comment:
Some of these flags are not really useful in a provider...  Not that I'm asking you to investigate as part of this PR, but I think we should at some point, so a TODO comment to that effect should be added.

Reply:
Found out the hard way that some of them are used :)